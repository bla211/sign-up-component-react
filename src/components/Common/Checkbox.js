import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Checkbox extends Component {
    render(){
        const { isChecked, toggleChecked } = this.props;
        
        return(    
            <label class="checkbox-wrapper">
                <input type="checkbox" checked={ isChecked } onChange={ () => toggleChecked() }/>
                <span class="checkmark"></span>
            </label>
        )
    }
}

Checkbox.propTypes = {
    isChecked: PropTypes.bool,
    toggleChecked: PropTypes.string,
  };

export default Checkbox;