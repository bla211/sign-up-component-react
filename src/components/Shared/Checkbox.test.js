import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Checkbox from './Checkbox';

configure({ adapter: new Adapter() });

describe('Test Checkbox component', () => {
    test('Checkbox renders properly when passed "isChecked === true"', () => {
        const isChecked = true;
        const wrapper = shallow(<Checkbox isChecked={isChecked}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Checkbox renders properly when passed "isChecked === false"', () => {
        const isChecked = false;
        const wrapper = shallow(<Checkbox isChecked={isChecked}/>);
        expect(wrapper).toMatchSnapshot();
    });
});