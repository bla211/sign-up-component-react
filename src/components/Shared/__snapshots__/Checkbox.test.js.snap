// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkbox component Checkbox renders properly when passed "isChecked === false" 1`] = `
<label
  className="checkbox-wrapper"
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkmark "
  />
</label>
`;

exports[`Test Checkbox component Checkbox renders properly when passed "isChecked === true" 1`] = `
<label
  className="checkbox-wrapper"
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkmark "
  />
</label>
`;
